.github/workflows/algolia/src/syncAlgoliaIndex.ts

```diff
@@ -0,0 +1,130 @@
+import fs from "fs";
+import path from "path";
+import matter from "gray-matter";
+import algoliasearch from "algoliasearch";
+
+type DocsFrontMatter = {
+  content: string;
+  data: {
+    objectId: string;
+    title?: string;
+    description?: string;
+    slug?: string;
+    tags?: Array<string>;
+  };
+};
+
+const getAllFiles = (
+  dirPath,
+  arrayOfFiles: Array<{ inputPath: string; outputPath: string; type: string }>,
+  type
+) => {
+  const files = fs.readdirSync(dirPath);
+
+  arrayOfFiles = arrayOfFiles || [];
+
+  files.forEach(function (file) {
+    if (fs.statSync(dirPath + "/" + file).isDirectory()) {
+      arrayOfFiles = getAllFiles(dirPath + "/" + file, arrayOfFiles, type);
+    } else {
+      const inputPath = path.join(dirPath, "/", file);
+      const outputPath = `.\\output\\${path.join(
+        "./output/",
+        dirPath,
+        "/",
+        file
+      )}`;
+      arrayOfFiles.push({ inputPath, outputPath, type });
+    }
+  });
+
+  return arrayOfFiles;
+};
+
+type AlgoliaRecord = {
+  objectId: any;
+  content: string;
+  objectID: string;
+  type: any;
+  title?: string;
+  description?: string;
+  slug?: string;
+  tags?: Array<string>;
+};
+
+const generateAlgoliaJson: () => AlgoliaRecord[] = () => {
```

Comment:
If the return type is `AlgoliaRecord[]` how about renaming the function to `generateAlgoliaRecords`?

```ts
const generateAlgoliaRecord: (): AlgoliaRecord[] => {
 // ...
}
```

When we see JSON, our brains can’t immediately tell if this is some object on an array. When there is plural, that’s a solid hit that we’ll have an array. Same with `xyzLookup` or `xyzById`: we don’t need to hover over the var to make a good judgement of its inner shape.